src/hotspot/share/gc/shared/gcLocker.hpp

```diff
@@ -27,126 +27,30 @@
 
 #include "gc/shared/gcCause.hpp"
 #include "memory/allStatic.hpp"
-#include "utilities/globalDefinitions.hpp"
-#include "utilities/macros.hpp"
-
-class JavaThread;
-
-// The direct lock/unlock calls do not force a collection if an unlock
-// decrements the count to zero. Avoid calling these if at all possible.
+#include "runtime/mutex.hpp"
 
 class GCLocker: public AllStatic {
- private:
-  // The _jni_lock_count keeps track of the number of threads that are
-  // currently in a critical region.  It's only kept up to date when
-  // _needs_gc is true.  The current value is computed during
-  // safepointing and decremented during the slow path of GCLocker
-  // unlocking.
-  static volatile jint _jni_lock_count;  // number of jni active instances.
-  static volatile bool _needs_gc;        // heap is filling, we need a GC
-  static uint _total_collections;        // value for _gc_locker collection
+  static Monitor* _lock;
+  static volatile bool _is_gc_request_pending;
 
 #ifdef ASSERT
-  // This lock count is updated for all operations and is used to
-  // validate the jni_lock_count that is computed during safepoints.
-  static volatile jint _debug_jni_lock_count;
+  static uint64_t _debug_count;
```

Comment:
Maybe the variable could be named something less generic, indicating what it is counting. Or add a comment.